tools/shared.py

```diff
@@ -927,12 +936,14 @@ def get_llvm_target():
 
 if LLVM_TARGET == WASM_TARGET:
   # wasm target does not automatically define emscripten stuff, so do it here.
-  COMPILER_OPTS = COMPILER_OPTS + ['-DEMSCRIPTEN',
-                                   '-D__EMSCRIPTEN__',
+  COMPILER_OPTS = COMPILER_OPTS + ['-D__EMSCRIPTEN__',
                                    '-Dunix',
                                    '-D__unix',
                                    '-D__unix__']
 
+  # The preprocessor define EMSCRIPTEN is deprecated. Don't pass it to code in strict mode. Code should use the define __EMSCRIPTEN__ instead.
+  if not is_emscripten_strict(): COMPILER_OPTS += ['-DEMSCRIPTEN']
```

Comment:
i think we discussed this already - this code happens before the `STRICT` setting is applied. so only the env var can influence this I think? ditto below.